frontend/src/api/RestaurantApi.tsx

```diff
@@ -0,0 +1,37 @@
+import { RestaurantResponse } from '@/types';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useQuery } from 'react-query';
+
+const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+export const useSearchRestaurants = (city?: string) => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
+    const accessToken = await getAccessTokenSilently();
+
+    const response = await fetch(
+      `${API_BASE_URL}/api/restaurant/search/${city}`,
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/37#issuecomment-2405201956>#67cbd7</a><span> - 10/10/2024, 02:13 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'city' parameter is not encoded before being used in the API URL, which could lead to potential security vulnerabilities.
</div>
<div id="fix">
<b>Fix:</b> Encode the 'city' parameter using encodeURIComponent() before including it in the API URL
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -11,7 +11,7 @@
      const accessToken = await getAccessTokenSilently();
 
      const response = await fetch(
 -      `${API_BASE_URL}/api/restaurant/search/${city}`,
 +      `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}`,
        {
          method: 'GET',
          headers: {
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged